src/abi.js

```diff
@@ -268,8 +268,20 @@ module.exports = class ABI {
 
   static load(name, file) {
     let data = JSON.parse(fs.readFileSync(file))
-    return Array.isArray(data)
-      ? new ABI(name, file, immutable.fromJS(data))
-      : new ABI(name, file, immutable.fromJS(data.abi))
+
+    let abi = null
```

Comment:
You could write this as
```js
let abi = Array.isArray(data)
  ? data
  : data.abi !== undefined
    ? data.abi
    : data.compilerOutput !== undefined && data.compilerOutput.abi !== undefined
      ? abi.compilerOutput.abi
      : undefined

if (abi === undefined) {
  throw Error(`Could not extract ABI from ${file}`)
}
```

That's what I would do to avoid the awkward `let abi = null` assignment upfront. But not everyone likes ternary operators...

Reply:
I'm not so used to that yet so I'll leave it as ifs.

Reply:
That's ok 👍